File: src/app/model/Purchase.ts
import { TransactionStatus } from './enums/TransactionStatus';
import { Policy } from './Policy';
import { User } from './User';
export interface Purchase {
    id?: number;
    user: User;
    policy: Policy;
    transactionId: string;
    amount: number;
    status: TransactionStatus;
    createdAt: string;
}

File: src/main/java/com/sni/insurancecompany/insurancecompany/model/Purchase.java
package com.sni.insurancecompany.insurancecompany.model;

import com.sni.insurancecompany.insurancecompany.model.enums.TransactionStatus;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Data
public class Purchase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    @ManyToOne
    @JoinColumn(name = "policy_id", nullable = false)
    private Policy policy;
    @Column(name = "transaction_id", nullable = false)
    private String transactionId;
    @Column(nullable = false)
    private Double amount;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TransactionStatus status;
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}
